### DISCUSSION TITLE
Not working for webos or tizen?

### DISCUSSION THREAD

**Discussion by thepunisherxx** [flexn-io/create#155] [General]
I have successfully installed renative with create extension and it works quite well for the web. When building for webos, I had to install engine-rn-web. After resolving some issues, it still doesn't work.

```
[ error ] [build] Failed to compile.

TS2307: Cannot find module '../platformAssets/runtime/logo.png' or its corresponding type declarations.
    21 | import '../platformAssets/runtime/fontManager';
    22 | import { StaticTheme, Theme } from './types';
  > 23 | import ICON_LOGO from '../platformAssets/runtime/logo.png';
       |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    24 |
    25 | export const hasMobileWebUI = isFactorMobile && isEngineRnNext;
    26 | export const hasHorizontalMenu = !isFactorMobile && !isFactorDesktop && !hasMobileWebUI;
```

**Comment by aurimasmi** [comment]
Are you trying to run template-starter?

**Reply by thepunisherxx** [reply]
Indeed, I'm encountering some issues with the template-starter as well. I am using a fresh installation of the latest renative version. (0.37.3)

```
[ error ] [build] Failed to compile.
TS2556: A spread argument must either have a tuple type or be passed to a rest parameter.
    2 | import App from '../app';
    3 |
  > 4 | const Main = (...args) => Launch(App, ...args);
      |                                       ^^^^^^^
    5 |
    6 | export default Main;
    7 |
```

**Reply by aurimasmi** [reply]
@wwwebgainss0 please try this combination:

rnv version `0.37.3-rc.1`
@flexn/create-template-starter version `1.1.0-alpha.6`

it has now default engine as engine-rn-web which gives to full code unification with other platforms, although it's noticeably slower than engine-lng which will come soon

**Reply by thepunisherxx** [reply]
Great, the combination worked perfectly for the Tizen build. Thank you! Now, when building for webOS, I get the message below. It can't find the index.html, which is not in the root but in the public folder. Am I missing something here?

Thank you for your assistance.
 
```
│ [ error ] [build] COMMAND:

C:\webOS_TV_SDK\CLI\bin\ares-package.cmd -o platformBuilds\app_webos\output platformBuilds\app_webos -n

FAILED with ERROR:

Command failed with exit code 4294967295: C:\webOS_TV_SDK\CLI\bin\ares-package.cmd -o platformBuilds\app_webos\output platformBuilds\app_webos -n
ares-package ERR! [Tips]: The specified path does not exist <platformBuilds\app_webos\index.html>

ares-package ERR! [Tips]: The specified path does not exist <platformBuilds\app_webos\index.html>

Error: Command failed with exit code 4294967295: webOS_TV_SDK\CLI\bin\ares-package.cmd -oplatformBuilds\app_webos\output platformBuilds\app_webos -n
ares-package ERR! [Tips]: The specified path does not exist <platformBuilds\app_webos\index.html>
    at makeError (node_modules\rnv\node_modules\execa\lib\error.js:60:11)
    at handlePromise (node_modules\rnv\node_modules\execa\index.js:118:26)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)

```

**Reply by aurimasmi** [reply]
What full rnv command are you running? What scheme?

**Reply by thepunisherxx** [reply]
I've tried executing both "rnv export -p webos" and "rnv build -p webos". While these commands work for Tizen, they don't seem to function for webOS. I'm encountering the error mentioned above.

**Reply by aurimasmi** [reply]
Alright I see there are some problems on renative side which needs to be fixed, I'll address this to developers as a high priority.

What you can try to do meanwhile is to add missing index.html file:

1. Create folder in `appConfigs/base/assets/webos`
2. Add index.html file from here https://github.com/flexn-io/renative/tree/main/packages/engine-rn-web/templates/platforms/webos/public to that newly created folder and try again

**Reply by thepunisherxx** [reply]
Hi @aurimasmi 
Thank you for your prompt response and support.

I've now created this folder at the specified path (appConfigs/base/assets/webos) and copied the index file into it.

Unfortunately, I'm still receiving the same error message during deployment, stating that the index cannot be found.

**Reply by thepunisherxx** [reply]
I have now copied the index.html to the webos Platformbuilds folder. Now I can compile the DemoTemplate. When I upload it to the emulator or TV device, I only get a black screen and I don't see any messages in the debug. . However, it still works correctly in the web browser.

**Reply by pauliusguzas** [reply]
Hello, this is known Renative bug and is being pushed to fix https://github.com/flexn-io/renative/issues/925. Though you can open debuger this way: `ares-inspect --device LG_webOS_TV --app com.mycompany.startertest1 --open`.  LG_webOS_TV should be changed to your device name and com.mycompany.starter1 should be changed to your app id
